### ISSUE TITLE
Can't install last few versions with VS 2022

### ISSUE THREAD

**Issue by KernelLeak** [agileobjects/ReadableExpressions#127]
I tried installing the latest version (4.5.0) for Visual Studio 2022, but the dialog at the end of the installation always looks like this:

```
---------------------------
Complete
---------------------------
Installed visualizers for:
---------------------------
OK   
---------------------------
```

which obviously hints at the installer not recognizing my installation of VS 2022.

I've downloaded all versions from 4.4.0 back to 4.0.0, but the only version that installs correctly is 4.0.0:

```
---------------------------
Complete
---------------------------
Installed visualizers for:

 - Visual Studio 2022
---------------------------
OK   
---------------------------
```

All later versions can't seem to find my installation of VS 2022 at ```c:\devel\vs2022``` - could this be the issue? But why would a custom installation path work in 4.0.0, but fail in all versions that came after it?

**Comment by SteveWilkes** [comment]
Hi,

Thanks for raising this, and for the effort of going back to previous versions and finding out when it stopped working!

I think the only way you can wind up with an empty 'Installed visualizers for:' list is if the installer finds 1 or more VS installations, but then fails to install any of them. The only way a visualizer installation can silently fail is if the expected visualizers directory doesn't exist, and it figures that out based on the VS install directory.

It gets VS install directories from the registry, looking in `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio*\Capabilities` > `ApplicationDescription`, then substrings that value to get the path to the 'IDE' directory.

So! It would be helpful if you could look in your registry and let me know what you have in the `ApplicationDescription` key for your VS2022 installation - it'll be inside a `VisualStudio_[RandomString]\Capabilities` key. Thanks!

**Comment by KernelLeak** [comment]
Sorry for taking my time to reply, but this problem concerns my developer machine at work which I didn't have access to over the weekend...

The value of ```ApplicationDescription``` I have under ```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio_94ec1717\Capabilities``` is ```@C:\Devel\VS2022\Common7\IDE\devenvdesc.dll,-1002``` - does that help?

**Comment by SteveWilkes** [comment]
Thanks very much! That'll be useful, yeah - I'll look into it and see if I can see what's going on (or isn't going on, in this case)

**Comment by KernelLeak** [comment]
> Ok, the installer is expecting there to be a directory at `C:\Devel\VS2022\Common7\Packages\Debugger\Visualizers` - does that exist?

Yes, it does - and with 4.0.0 there's even a ```AgileObjects.ReadableExpressions.Visualizers.Vs17.dll``` in there...

**Comment by KernelLeak** [comment]
I guess I found it - looking at the source code for the custom action, the only thing that changed was the way the VS version is determined:

4.0.0 parsed the the ApplicationName from the registry, whereas all following versions parse the directories in the installation path as integers, so of course my "VS2022" isn't recognized...

**Comment by SteveWilkes** [comment]
That's it! Thanks! I think `ApplicationName` turned out not be reliable in some scenarios so I switched to the install path - I've updated it to fall back to `ApplicationName` if it can't find a version in the path.

Release to follow! Thanks again!

**Comment by KernelLeak** [comment]
That seems like it'd be best of both worlds - thanks! :)

**Comment by SteveWilkes** [comment]
This is (finally) fixed in the v4.6 installer, which is now [on VS Marketplace](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1232914.ReadableExpressionsVisualizers). Thanks!
